sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -2475,18 +2476,21 @@ checkDevSupportDeviceRequirements(const device &Dev,
   }
 
   // Checking if device supports defined required work group size
-  if (ReqdWGSizePropIt) {
-    ByteArray ReqdWGSize =
-        DeviceBinaryProperty(*(ReqdWGSizePropIt.value())).asByteArray();
+  if (ReqdWGSizeUint32TPropIt || ReqdWGSizeSizeTPropIt) {
+    bool usingSizeT = ReqdWGSizeSizeTPropIt.has_value();
+    auto it = usingSizeT ? ReqdWGSizeSizeTPropIt : ReqdWGSizeUint32TPropIt;
+
+    ByteArray ReqdWGSize = DeviceBinaryProperty(*(it.value())).asByteArray();
     // Drop 8 bytes describing the size of the byte array.
     ReqdWGSize.dropBytes(8);
-    uint64_t ReqdWGSizeAllDimsTotal = 1;
-    std::vector<uint64_t> ReqdWGSizeVec;
+    size_t ReqdWGSizeAllDimsTotal = 1;
+    std::vector<size_t> ReqdWGSizeVec;
     int Dims = 0;
     while (!ReqdWGSize.empty()) {
       // The reqd_work_group_size data is stored as uint32_t's,
```

Comment:
The comment should be updated that only old binaries store `reqd_work_group_size` as 32-bit.
We should also put `TODO`s to both runtime and the compiler to remove that extra `_size_t` property once ABI break is allowed again